### PR TITLE
[HW5] (Application architecture)

### PR DESCRIPTION
**Pull request by AlwaysSayNo**
- Added general and extended diagrams.
- General diagram shows the general concepts of modules and their communication.
- Extended diagram shows communication among specific modules and more detailed infrastructure of application.

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -49,6 +61,28 @@ func main() {
...
+	jsDelivrClient, err := cdnjsdelivr.NewClient(config.LoadCurrencyServiceConfig())
+	if err != nil {
+		log.Fatal("creating JS Deliver currency provider")
+	}
+
+	// SET PROVIDERS CHAIN
+	govUaClient.SetNext(jsDelivrClient)
+	privateClient.SetNext(govUaClient)
+
+	return privateClient
```
Comment (andiserg) [inline cmd/main.go]: what does this function do in cmd/main.go?

File: internal/service/user/service.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	if s.userRepository.ExistsByEmail(userModel.Email) {
+		return nil, apperrors.NewUserWithEmailExistsError()
+	}
+
+	savedUser, err := s.userRepository.Create(userModel)
+	if err != nil {
+		return nil, fmt.Errorf("saving savedUser in database: %w", err)
+	}
+
+	userDTO := dto.ToDTO(*savedUser)
```
Comment (andiserg) [inline internal/service/user/service.go]: It looks strange. A Data Transfer Object is a simple data structure that does not require knowledge of how to map other structures to it

Comment (AlwaysSayNo) [reply internal/service/user/service.go]: dto in this example is a package, not an object

File: cmd/main.go (hunk 1 above)
Comment (AlwaysSayNo) [reply cmd/main.go]: This function setups raters chain

Comment (andiserg) [reply cmd/main.go]: In my opinion, it should be placed in the currency module

**Pull request merged**